tests/unit_test_smtp_credentials_check.py

```diff
@@ -0,0 +1,94 @@
+            self.test_expecting_SMTP_connection_error.__name__, check)
+        assert (status, message) == (
+            False, smtpcheck.Messages.SMTP_CONNECTION_ERROR.value)
+
+    def test_expecting_general_connection_refused_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(ConnectionRefusedError())
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_general_connection_refused_error.__name__, check)
+        assert (status, message) == (
+            False, smtpcheck.Messages.CONNECTION_REFUSED_ERROR.value)
+
+    def test_expecting_SMTP_helo_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(
+            smtplib.SMTPHeloError(0, ''))
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_SMTP_helo_error.__name__, check)
+        assert (status, message) == (
+            False, smtpcheck.Messages.HELO_OR_EHLO_ERROR.value)
+
+    def test_expecting_SMTP_server_disconnected_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(
+            smtplib.SMTPServerDisconnected())
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_SMTP_server_disconnected_error.__name__, check)
+        assert (status, message) == (
+            False, smtpcheck.Messages.DISCONNECTION_ERROR.value)
+
+    def test_expecting_SMTP_authentication_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(
+            smtplib.SMTPAuthenticationError(0, ''))
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_SMTP_authentication_error.__name__, check)
+        assert (status, message) == (
+            False, smtpcheck.Messages.AUTHENTICATION_ERROR.value)
+
+    def test_expecting_general_SMTP_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(smtplib.SMTPException())
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_general_SMTP_error.__name__, check)
+        assert (status, message) == (
+            False, smtpcheck.Messages.UNKNOWN_SMTP_ERROR.value)
+
+    def test_expecting_general_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(Exception())
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_general_error.__name__, check)
+        assert (status, message) == (False, smtpcheck.Messages.UNKNOWN_ERROR.value)
+    
+    def test_expecting_auth_command_not_supported_error(self) -> None:
+        check = TestSmtpCredentials._create_mocked_method_raising(smtplib.SMTPNotSupportedError())
+        status, message = TestSmtpCredentials._run_mocked_check(
+            self.test_expecting_auth_command_not_supported_error.__name__, check)
+
+        self.assertEqual(
```

Comment:
If you would use self.assertEqual instead of assert, then the messages would be more detailed.
assertEqual will show current string and expected string. assert is a language syntax and will show only that the line does not equal True.

assertEqual is from `unittest`, `assert` is from Python natively.